subsys/net/lib/conn_mgr/conn_mgr_connectivity.h

```diff
@@ -0,0 +1,325 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief TODO
+ */
+
+#ifndef ZEPHYR_INCLUDE_CONN_MGR_CONNECTIVITY_H_
+#define ZEPHYR_INCLUDE_CONN_MGR_CONNECTIVITY_H_
+
+#include <zephyr/device.h>
+#include <zephyr/net/net_if.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief TODO: DESCRIBE
+ * @defgroup conn_mgr_connectivity TODO: DESCRIBE
+ * @ingroup networking TODO: GROUP
+ * @{
+ */
+
+/**
+ * @brief Network L2 connectivity API structure
```

Comment:
Should we still call this L2 connectivity? I'm not sure if there is a better name though.